velox/dwio/parquet/reader/ParquetData.h

```diff
@@ -196,6 +207,38 @@ class ParquetData : public dwio::common::FormatData {
     return true;
   }
 
+  void setBloomFilterInputStream(
+      uint32_t rowGroupId,
+      dwio::common::BufferedInput& bufferedInput) {
+    if (bloomFilterInputStream_ != nullptr) {
+      return;
+    }
+    auto rowGroup = fileMetaDataPtr_.rowGroup(rowGroupId);
+    auto colChunk = rowGroup.columnChunk(type_->column());
+
+    if (!colChunk.hasBloomFilterOffset()) {
+      return;
+    }
+
+    VELOX_CHECK(
+        !colChunk.hasCryptoMetadata(),
+        "Cannot read encrypted bloom filter yet");
+
+    auto bloomFilterOffset = colChunk.bloomFilterOffset();
+    auto fileSize = bufferedInput.getInputStream()->getLength();
+    VELOX_CHECK_GT(
+        fileSize,
+        bloomFilterOffset,
+        "file size {} less or equal than bloom offset {}",
+        fileSize,
+        bloomFilterOffset);
+
+    bloomFilterInputStream_ = bufferedInput.read(
```

Comment:
This is doing the IO one by one and not efficient.  You need to use `enqueue` here and call `bufferedInput.load()` once after all the regions are enqueued in `filterRowGroups`.